test/parallel_api/numeric/numeric.ops/reduce_by_segment.pass.cpp

```diff
@@ -351,10 +351,10 @@ main()
         test_flag_pred<sycl::usm::alloc::device, class KernelName8, dpl::complex<float>>();
 #endif // TEST_DPCPP_BACKEND_PRESENT
 
-#if !_PSTL_ICC_TEST_SIMD_UDS_BROKEN
+#if !_PSTL_ICC_TEST_SIMD_UDS_BROKEN && !_PSTL_ICPX_TEST_RED_BY_SEG_OPTIMIZER_CRASH
```

Comment:
From technical point of view we are able to pack all this logic into definition of ```_PSTL_ICC_TEST_SIMD_UDS_BROKEN``` without changes in this file